### PR TITLE
Remove deprecated unique function

### PR DESCRIPTION
**Pull request by MartyEwings**
unique function has been deprecated in favour of core version since puppet 5

### CODE REVIEW

File: lib/puppet/parser/functions/unique.rb
Comment (alexjfisher) [inline lib/puppet/parser/functions/unique.rb:27]: This is intriguing.  Was this ever reachable??  ie If you were on Puppet 5 wouldn't the built in function automatically get used instead and was there anyway round this?

**Review by LukasAud — APPROVED** [review]
LGTM. Will wait for a second review before merging tho.

**Review by b4ldr — APPROVED** [review]

**Review by chelnak — APPROVED** [review]

**Pull request merged**
